presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1305,6 +1307,13 @@ else if (sourceNode instanceof SampleNode) {
                     return new PhysicalOperation(operatorFactory, outputMappings, context, source);
                 }
             }
+            catch (PrestoException e) {
+                if (e.getErrorCode() == COMPILER_ERROR.toErrorCode() && getExceptionRootCause(e) instanceof MethodTooLargeException) {
+                    throw new PrestoException(COMPILER_ERROR,
+                        "Query exceeded maximum columns or filters. Please reduce the number of columns and filters referenced and re-run the query.", e);
```

Comment:
Nit: the second sentence should also say "columns or filters" as the user might not need to reduce both

Reply:
@electrum "method too large" should be an internal- or user- error?

Reply:
Let’s leave the classification for now. We don’t have specific limits here and it’s not exactly under user control.